### DISCUSSION TITLE
Device name change dialog - "NO" anyway change the name

### DISCUSSION THREAD

**Discussion by Masterz69** [home-assistant/frontend#8645] [Other feature requests] [answered]
Just checked on 2021.3.4 - if press "NO" in device name change dialog: it changes name anyway :)

**Comment by spacegaier** [comment, marked as answer]
I cannot reproduce this. Works just fine for me. But you might be talking about something else, because I do not see a "No" button in the device rename dialog. Perhaps you can add a screenshot to illustrate.

**Reply by Masterz69** [reply]
![image](https://user-images.githubusercontent.com/15225823/111074950-2dc68580-84ee-11eb-8009-add7e4556886.png)

NO on the left, RENAME on the right.


Ohh, now I see - this is NO about renaming Entities. But who reads so much letters ? :)
Probably button CANCEL would be good to stop renaming device.

Thank You very much.

**Reply by Masterz69** [reply]
There is no way back if rename Device initiated. Pressing ESC not cancelling process.

**Reply by spacegaier** [reply]
At this point in the process we cannot stop the device renaming since that part took already place. Only if that first step worked, will we show the popup that offers to rename the entity IDs accordingly.

In the past the button said actually "Cancel" but that was confusing/wrong since it does not cancel, which is why I changed it to "'No" some time ago.

We could perhaps switch out "rename" with "adjust", but still: If people do not read the text (that I on purpose made yellow to highlight the warning nature), then there is not much we can do :) .

**Reply by spacegaier** [reply]
I proposed a wording change here: https://github.com/home-assistant/frontend/pull/8648

**Reply by BoneheadFraggle** [reply]
Thinking about it, you could argue that the warning comes a bit late.

You may think you want to rename the device and click Update.
But by the time you actually get the yellow warning you realise it's a bad idea to rename at all.
Then there's no way back.

I think it would be an improvement to make the steps and consequences more clear from the start.

One way could be by putting the yellow text already in the first dialog. 

Another way to introduce another dialog with a possibility to cancel before the "Do you also.."

Or change the logic to 
1. Click Update
2. Choose from "Cancel/Abort/I changed my mind", "No" and "Update"

so there is a way to actually abort the procedure when you realise what is about to happen.

A small note: 
Yellow text on white background is really hard to read. You may want to consider another color.
In dark mode it is great, but not everyone uses dark mode (as in the example above).
